galleries/examples/subplots_axes_and_figures/subfig_mosaic.py

```diff
@@ -0,0 +1,399 @@
+fig = plt.figure()
+subfigs = fig.subfigure_mosaic(mosaic)
+identify_subfigs(subfigs)
+
+# %%
+# SubFigures spanning multiple rows/columns
+# =========================================
+#
+# Something we can do with `.Figure.subfigure_mosaic`, that we cannot
+# do with `.Figure.subfigures`, is to specify that a SubFigure should span
+# several rows or columns.
+
+
+# %%
+# If we want to re-arrange our four SubFigures to have ``"C"`` be a horizontal
+# span on the bottom and ``"D"`` be a vertical span on the right we would do:
+
+subfigs = plt.figure().subfigure_mosaic(
+    """
+    ABD
+    CCD
+    """
+)
+
+# setting edges for clarity
+for sf in subfigs.values():
+    sf.set_edgecolor('black')
+    sf.set_linewidth(2.1)
+
+identify_subfigs(subfigs)
+
+# %%
+# If we do not want to fill in all the spaces in the Figure with SubFigures,
+# we can specify some spaces in the grid to be blank, like so:
+
+subfigs = plt.figure().subfigure_mosaic(
+    """
+    A.C
+    BBB
+    .D.
+    """
+)
+
+# setting edges for clarity
+for sf in subfigs.values():
+    sf.set_edgecolor('black')
+    sf.set_linewidth(2.1)
+
+identify_subfigs(subfigs)
+
+# %%
+# If we prefer to use another character (rather than a period ``"."``)
+# to mark the empty space, we can use *empty_sentinel* to specify the
+# character to use.
+
+subfigs = plt.figure().subfigure_mosaic(
+    """
+    aX
+    Xb
+    """,
+    empty_sentinel="X",
+)
+
+# setting edges for clarity
+for sf in subfigs.values():
+    sf.set_edgecolor('black')
+    sf.set_linewidth(2.1)
+identify_subfigs(subfigs)
+
+# %%
+#
+# Internally there is no meaning attached to the letters we use, any
+# Unicode code point is valid!
+
+subfigs = plt.figure().subfigure_mosaic(
+    """αя
+       ℝ☢"""
+)
+
+# setting edges for clarity
+for sf in subfigs.values():
+    sf.set_edgecolor('black')
+    sf.set_linewidth(2.1)
+identify_subfigs(subfigs)
+
+# %%
+# It is not recommended to use white space as either a label or an
+# empty sentinel with the string shorthand because it may be stripped
+# while processing the input.
+#
+# Controlling mosaic creation
+# ===========================
+#
+# This feature is built on top of `.gridspec` and you can pass the
+# keyword arguments through to the underlying `.gridspec.GridSpec`
+# (the same as `.Figure.subfigures`).
+#
+# In this case we want to use the input to specify the arrangement,
+# but set the relative widths of the rows / columns.  For convenience,
+# `.gridspec.GridSpec`'s *height_ratios* and *width_ratios* are exposed in the
+# `.Figure.subfigure_mosaic` calling sequence.
+
+subfigs = plt.figure().subfigure_mosaic(
+    """
+    .a.
+    bAc
+    .d.
+    """,
+    # set the height ratios between the rows
+    height_ratios=[1, 3.5, 1],
+    # set the width ratios between the columns
+    width_ratios=[1, 3.5, 1],
+)
+
+# setting edges for clarity
+for sf in subfigs.values():
+    sf.set_edgecolor('black')
+    sf.set_linewidth(2.1)
+identify_subfigs(subfigs)
+
+# %%
```

Comment:
The subplot_mosaic [example](https://matplotlib.org/devdocs/users/explain/axes/mosaic.html#controlling-mosaic-creation) also has a version of this using gridspec_kw. But for subfigures the following analogous code:
``` python3
mosaic = """AA
            BC"""
fig = plt.figure()
subfigs = fig.subfigure_mosaic(
    mosaic,
    subfigure_kw={'edgecolor':'black','linewidth':2},
    gridspec_kw={
        "bottom": 0.25,
        "top": 0.95,
        "left": 0.1,
        "right": 0.5,
        "wspace": 0.5,
        "hspace": 0.5,
    },
)
identify_subfigs(subfigs)

subfigs = fig.subfigure_mosaic(
    mosaic,
    subfigure_kw={'edgecolor':'black','linewidth':2},
    gridspec_kw={
        "bottom": 0.05,
        "top": 0.75,
        "left": 0.5,
        "right": 1,
        "wspace": 0.5,
        "hspace": 0.5,
    },
)
identify_subfigs(subfigs)
```
gives this:
<img src="https://github.com/matplotlib/matplotlib/assets/100782385/e6552c8a-986b-4f68-846a-8466021711a2" width="240" height="240">
when it should look like this:
<img src="https://github.com/matplotlib/matplotlib/assets/100782385/a29f6714-7cbe-483e-9e4a-e71fcc03f99e" width="240" height="240">
So the only gridspec_kws that are actually being taken into account (if specified) are width/height ratios.